simulation_data.R

```diff
@@ -2,161 +2,109 @@
-  Sigma <- Exponential(d, range=Theta, phi=Phi)
-  W <- mvrnorm(n=1, mu=rep(0, length(cells.all)), Sigma)
-  N <- length(W)
-  
-  #### Simulate locations
-  locs <- simLocations(caPr.disc, beta.loc, w=W)
-  obs_cells <- c(obs_cells, sum(locs$status))
-  
-  ps_contribs <- c(ps_contribs, 
-                   calc_ps_contribution(caPr.disc, locs, beta.case, Alpha.case, beta.ctrl, Alpha.ctrl, W))
   
+  prevalences <- c()
+  ps_contribs <- c()
+  obs_cells <- c()
+  n_specimen <- c()
   
-  #### Simulate counts given locations
-  case.data <- simCounts(caPr.disc, locs, beta.case, Alpha.case, W)
-  ctrl.data <- simCounts(caPr.disc, locs, beta.ctrl, Alpha.ctrl, W)
-  prevalences <- c(prevalences, sum(case.data$y)/sum(case.data$y + ctrl.data$y))
-  n_specimen <- c(n_specimen, sum(case.data$y + ctrl.data$y))
-  
-  params <- list(
-    sampling="high",
-    beta.case=beta.case,
-    beta.ctrl=beta.ctrl,
-    alpha.case=Alpha.case,
-    alpha.ctrl=Alpha.ctrl,
-    beta.loc=beta.loc,
-    Theta=Theta,
-    Phi=Phi,
-    W=W
-  )
-  
-  if (dst != ""){
-    save_output(params, paste("params_high_", i, ".json", sep=""), dst=dst)
+  for (i in 1:n_sims) {
+    
+    #### Simulate Gaussian process
+    cells.all <- c(1:ncell(disc))[!is.na(values(disc[[1]]))]
+    coords <- xyFromCell(disc, cell = cells.all)
+    d <- as.matrix(dist(coords, diag = TRUE, upper = TRUE))
+    Sigma <- Exponential(d, range = Theta, phi = Phi)
+    W <- mvrnorm(n = 1, mu = rep(0, length(cells.all)), Sigma)
+    
+    #### Simulate locations
+    locs <- simLocations(disc, beta.loc, w = W, cells.all = cells.all, d = d)
+    obs_cells <- c(obs_cells, sum(locs$status))
+    
+    ps_contribs <- c(ps_contribs, 
+                     calc_ps_contribution(disc, locs, beta.case, Alpha.case, beta.ctrl, Alpha.ctrl, W))
+    
+    #### Simulate counts given locations
+    case.data <- simCounts(disc, locs, beta.case, Alpha.case, W)
+    ctrl.data <- simCounts(disc, locs, beta.ctrl, Alpha.ctrl, W)
+    prevalences <- c(prevalences, sum(case.data$y)/sum(case.data$y + ctrl.data$y))
+    n_specimen <- c(n_specimen, sum(case.data$y + ctrl.data$y))
+    
+    params <- list(
+      sampling = label,
+      beta.case = beta.case,
+      beta.ctrl = beta.ctrl,
+      alpha.case = Alpha.case,
+      alpha.ctrl = Alpha.ctrl,
+      beta.loc = beta.loc,
+      Theta = Theta,
+      Phi = Phi,
+      W = W
+    )
+    
     data <- list(
-      case.data=case.data,
-      ctrl.data=ctrl.data,
-      locs=locs
+      case.data = case.data,
+      ctrl.data = ctrl.data,
+      locs = locs
     )
-    save_output(data, paste("data_high_", i, ".json", sep=""), dst=dst)
+    
+    if (!is.null(save_dir)) {
+      dir.create(save_dir)
+      param_fname <- paste0("params_", label, "_", i, ".json")
+      data_fname  <- paste0("data_", label, "_", i, ".json")
+      
+      write( toJSON(params), file.path(save_dir, param_fname) )
+      write( toJSON(data), file.path(save_dir, data_fname) )
```

Comment:
I thought it'd be better to just explicitly write out the JSON save code, since it's not that much longer than the `save_output` helper function you have and makes it clearer what's happening. Writing out the full path for saving rather than separating it into the main directory, and then the filename also cuts out a an intermediary step.